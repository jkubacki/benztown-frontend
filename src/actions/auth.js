import { SubmissionError } from 'redux-form';
import { push } from 'connected-react-router';

import { dispatchRequest } from 'actions';
import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT_REQUEST,
  LOGOUT_SUCCESS,
  LOGOUT_FAILURE,
  MARK_AS_NOT_LOGGED,
  ACCEPT_INVITATION_REQUEST,
  ACCEPT_INVITATION_SUCCESS,
  ACCEPT_INVITATION_FAILURE
} from 'actionTypes';
import {
  sendLoginRequest,
  sendAcceptInvitationRequest,
  sendLogoutRequest,
} from 'api/auth';
import { getRootPath } from 'constants/paths';

export function logout() {
  return dispatchRequest({
    requestAction: LOGOUT_REQUEST,
    request: sendLogoutRequest,
    onSuccess: dispatch => {
      dispatch(push(getRootPath()));
      dispatch({ type: LOGOUT_SUCCESS });
    },
    onFailure: dispatch => {
      dispatch({ type: LOGOUT_FAILURE });
    },
  });
}

export function acceptInvitation(params) {
  return dispatchRequest({
    requestAction: ACCEPT_INVITATION_REQUEST,
    request: () => sendAcceptInvitationRequest(params),
    onSuccess: dispatch => {
      dispatch({ type: ACCEPT_INVITATION_SUCCESS });
      dispatch(push(getRootPath()));
    },
    onFailure: dispatch => {
      dispatch({ type: ACCEPT_INVITATION_FAILURE });
    },
  });
}

export function login(params) {
  return dispatchRequest({
    requestAction: LOGIN_REQUEST,
    request: () => sendLoginRequest(params),
    onSuccess: (dispatch, data, getState) => {
      dispatch({ type: LOGIN_SUCCESS, payload: data });
    },
    onFailure: (dispatch, error) => {
      dispatch({ type: LOGIN_FAILURE });

      if (error.response.data.error === 'invalid_grant') {
        throw new SubmissionError({
          email: 'errors.invalidLoginOrPassword',
          password: 'errors.invalidLoginOrPassword',
        });
      }
    },
  });
}

export function markAsLoggedIn() {
  return { type: LOGIN_SUCCESS };
}

export function markAsNotLoggedIn() {
  return { type: MARK_AS_NOT_LOGGED };
}
