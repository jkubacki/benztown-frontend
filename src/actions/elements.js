import dispatchRequest from 'actions';
import { requestElements } from 'api/elements';

import {
  REQUEST_ELEMENTS_FAILURE,
  REQUEST_ELEMENTS_REQUEST,
  REQUEST_ELEMENTS_SUCCESS,
} from 'actionTypes';

export default function getElements(params) {
  return dispatchRequest({
    requestAction: REQUEST_ELEMENTS_REQUEST,
    request: () => requestElements(params),
    onSuccess: (dispatch, payload) => {
      dispatch({ type: REQUEST_ELEMENTS_SUCCESS, payload: payload.data.data });
    },
    onFailure: (dispatch) => {
      dispatch({ type: REQUEST_ELEMENTS_FAILURE });
    },
  });
}
