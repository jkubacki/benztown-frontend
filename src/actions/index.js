function handleSuccess(dispatch, onSuccess, getState) {
  return (response) => {
    onSuccess(dispatch, response, getState);
    return true;
  };
}

function handleFailure(dispatch, onFailure, getState) {
  return (error) => {
    onFailure(dispatch, error, getState);
    return true;
  };
}

export default function dispatchRequest(
  {
    request, requestAction, onSuccess, onFailure,
  },
) {
  return (dispatch, getState) => {
    dispatch({ type: requestAction });

    return request(getState)
      .then(handleSuccess(dispatch, onSuccess, getState))
      .catch(handleFailure(dispatch, onFailure, getState));
  };
}
