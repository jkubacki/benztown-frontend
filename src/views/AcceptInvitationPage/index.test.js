import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { acceptInvitation } from 'actions/auth';
import { AcceptInvitationPageUnwrapped as AcceptInvitationPage } from './index';

jest.mock('actions/auth', () => ({
  acceptInvitation: jest.fn(() => Promise.resolve()),
}));

const props = {
  acceptInvitation,
  invitation_token: 'invitation_token',
  handleSubmit: jest.fn(x => x),
};

describe('AcceptInvitationPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AcceptInvitationPage {...props} />);
  });

  it('should render corectly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render action button', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('should render form', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render 1 Field Component', () => {
    expect(wrapper.find('Field').length).toEqual(1);
  });

  it('should render password', () => {
    expect(wrapper.find('Field[name="password"]').length).toEqual(1);
  });

  it('should call acceptInvitation with correct input values and with invitation token', () => {
    const instance = wrapper.instance();

    instance.handleSubmit({
      invitation_token: 'invitation_token',
      password: 'password',
    });

    expect(acceptInvitation).toHaveBeenCalledWith({
      invitation_token: 'invitation_token',
      password: 'password',
    });
  });
});
