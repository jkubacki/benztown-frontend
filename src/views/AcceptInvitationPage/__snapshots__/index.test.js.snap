// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptInvitationPage should render corectly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <div
    className="form-group"
  >
    <Field
      className="form-control"
      component={[Function]}
      label="password"
      name="password"
      placeholder="Set your password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    Accept Invitation
  </button>
</form>
`;
